### PR TITLE
[14.0][MIG] sale_order_revision 

### PR DESCRIPTION
**Pull request by kittiu**
From version 14 on, to use sale_order_revision, we will need base_revision + sale_order_revision.

Dependency,

- [x] base_revision https://github.com/OCA/server-ux/pull/246

### CODE REVIEW

**Review by Saran440 — COMMENTED** [review]
Functional Test
New revision display name is False-XX
Selection_002

May be you have to add
rec.flush()
in function create (base_revision)

**Comment by kittiu** [comment]
> Functional Test
> New revision display name is False-XX

What is your step to error, I can't replicate.

**Comment by Saran440** [comment]
Display name new revision will False from old document

Step to error
1. create SO document
2. install module sale_order_revision
3. click button new revision (Old SO)
4. display new document is False

**Comment by kittiu** [comment]
> Display name new revision will False from old document
> 
> Step to error
> 
> 1. create SO document
> 2. install module sale_order_revision
> 3. click button new revision (Old SO)
> 4. display new document is False

Thank you, I fixed.

**Review by Saran440 — APPROVED** [review]

**Comment by newtratip** [comment]
I can't install this module with new database, Error as following below.
Selection_862
Selection_863

I installed base_revision
Selection_864

This is my video
Peek 2021-01-07 16-51

@kittiu, Can you check ?

**Comment by Chandresh-OSI** [comment]
supersede with #1491

**Comment by kittiu** [comment]
> supersede with #1491

Do we need supercede?

**Comment by Chandresh-SerpentCS** [comment]
I just made one change with new PR.

**Comment by Chandresh-SerpentCS** [comment]
The module was not able to install in a new DB.

**Comment by kittiu** [comment]
> The module was not able to install in a new DB.

Thank you. You can also make comment to this PR. We can fix it.

File: sale_order_revision/__init__.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+
```
Comment (Chandresh-SerpentCS) [inline sale_order_revision/__init__.py:4]: Please add `from . import models` in this file.

File: sale_order_revision/view/sale_order.xml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record id="view_order_form" model="ir.ui.view">
```
Comment (Chandresh-SerpentCS) [inline sale_order_revision/view/sale_order.xml:3]: Make XML record ID unique, 
```suggestion
    <record id="view_order_form_inherit_revision" model="ir.ui.view">
```

File: sale_order_revision/view/sale_order.xml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        <field name="name">sale.order.form</field>
```
Comment (Chandresh-SerpentCS) [inline sale_order_revision/view/sale_order.xml:4]:
```suggestion
        <field name="name">sale.order.form.revision</field>
```

**Review by Chandresh-SerpentCS — CHANGES_REQUESTED** [review]

**Review by Chandresh-SerpentCS — CHANGES_REQUESTED** [review]
Add `server-ux` in oca_dependencies.txt

File: sale_order_revision/view/sale_order.xml (hunk 2 above)
Comment (kittiu) [reply sale_order_revision/view/sale_order.xml:3]: I think this one is ok. The full external ID will be `sale_order_revision.view_order_form` anyway.

File: sale_order_revision/__init__.py (hunk 1 above)
Comment (kittiu) [reply sale_order_revision/__init__.py:4]: Thank you for your review. Fixed.

**Comment by Chandresh-SerpentCS** [comment]
Conflict in one file.

**Comment by Chandresh-OSI** [comment]
Please fix the Travis of this PR.

**Comment by kittiu** [comment]
> Please fix the Travis of this PR.

Looks like problem is from sale_order_lot_generator. 
https://travis-ci.com/github/OCA/sale-workflow/jobs/487066492#L1567
Can you help fixing  that?

**Comment by newtratip** [comment]
> > Please fix the Travis of this PR.
> 
> Looks like problem is from sale_order_lot_generator.
> https://travis-ci.com/github/OCA/sale-workflow/jobs/487066492#L1567
> Can you help fixing that?

Fixed.

**Comment by NICO-SOLUTIONS** [comment]
Hey...
i am facing the following problem:
When i use the revision module and want to setup a new revision, it will not add a revision number to the base numer of the quotation. Instead it jumps to the next  base number of the quote. 

Adiitional informations:
- I use the module sale_quotation_number as well, to separate quotations from orders. 
- My sequence code prefix for quotations is "SQ/%(year)s/" and for orders "SO/%(year)s/"

So mabye one of this is the problem?

All the best

**Comment by rousseldenis** [comment]
> Hey...
> i am facing the following problem:
> When i use the revision module and want to setup a new revision, it will not add a revision number to the base numer of the quotation. Instead it jumps to the next base number of the quote.
> 
> Adiitional informations:
> 
> * I use the module sale_quotation_number as well, to separate quotations from orders.
> * My sequence code prefix for quotations is "SQ/%(year)s/" and for orders "SO/%(year)s/"
> 
> So mabye one of this is the problem?
> 
> All the best

Could you do an issue for that ?

**Comment by NICO-SOLUTIONS** [comment]
of course!
thanks

**Pull request merged**